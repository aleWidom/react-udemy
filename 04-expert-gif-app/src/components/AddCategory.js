import React, {useState} from 'react'
/* import PropTypes from 'prop-types'; */

//COMPONENTE DEBE TENER UNA UNICA TAREA ESPECIFICA Y SE OCUPE BIEN DE ELLA
const AddCategory = ({onNewCategory}) => {
    
    const [inputValue, setInputValue] = useState("")
    
    
    const changeValue = (e) => (
        setInputValue(e.target.value)
    )


    const submit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2) {
            onNewCategory(inputValue.trim())
            setInputValue("")
        }
    };



    return (
        <form onSubmit={submit}>
          <input type="text" value={inputValue} onChange={changeValue}/>
        </form>
   )
}


export default AddCategory;