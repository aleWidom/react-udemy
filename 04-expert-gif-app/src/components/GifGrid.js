import { GifGridItem } from './';
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({category}) => {

 const {data:images,loading} = useFetchGifs(category);
    
    return (
      <>
        <h3>{category}</h3>
        {loading && <p>Loading</p>}
        <div className="card-grid">
               { images.map((e)=> (
                    <GifGridItem key={e.id} {...e}
                    //otra forma de crear las props en vez de ponerlo como argumento con un nombre mando un objeto, antes mandaba por ej data={e}, entonces cuando lo utilizaba tenía que desetructurar el elemento o llamar las props con lo que traía dentro por separado, ahora de esta manera desestructuro cada propiedad y listo.
                    />   
                ))}
        </div>
      </>
    )


}
