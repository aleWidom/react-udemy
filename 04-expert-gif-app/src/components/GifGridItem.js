import React from 'react'

export const GifGridItem = ({id,title,url}) => {

    
    /*
    esto es si hubiese utilizado en el componente padre cuando utilizo este componente
    de esta manera

    console.log(props.data.id)
    console.log(props.data.url)
    console.log(props.data.title)
    
    { images.map((e)=> (
                    <GifGridItem key={e.id}  data={e}
    />   
    ))}
    */


    return (
        <div className="card animate__animated animate__bounce">
            <h2>{title}</h2>
            <img src={url} alt={title}/>
        </div>
    )
}
