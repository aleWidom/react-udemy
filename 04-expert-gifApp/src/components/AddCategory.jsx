import { useState } from 'react'
import PropTypes from 'prop-types';

//COMPONENTE DEBE TENER UNA UNICA TAREA ESPECIFICA Y SE OCUPE BIEN DE ELLA
export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState("")


    const changeValue = (e) => (
        setInputValue(e.target.value)
    )


    const submit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return
        onNewCategory(inputValue.trim())
        setInputValue("")
    };

    return (
        <form onSubmit={submit} aria-label='form'>
            <input type="text" value={inputValue} onChange={changeValue} placeholder={'Buscar Gifs'} />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
