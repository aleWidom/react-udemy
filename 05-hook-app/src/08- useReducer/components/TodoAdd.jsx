import { useFormTodos } from "../../hooks/useFormTodos"

export const TodoAdd = ({ todos }) => {

    const { input, onInputChange, onSubmitTodos } = useFormTodos(todos)

    return (
        <form>
            <input type='text' placeholder='Que hay que hacer?' value={input} onChange={onInputChange} />
            <button onClick={onSubmitTodos} type='submit'>Agregar</button>
        </form>
    )
}
