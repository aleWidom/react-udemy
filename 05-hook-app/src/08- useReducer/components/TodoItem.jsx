


export const TodoItem = ({ description }) => {
    return (
        <li >
            <span>{description}</span>
            <button>Borrar</button>
        </li>
    )
}
