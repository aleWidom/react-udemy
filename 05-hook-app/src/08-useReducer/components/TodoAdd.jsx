import { useFormTodos } from "../../hooks/useFormTodos"

export const TodoAdd = ({ handleNewTodo }) => {

    const { input, onInputChange, onSubmitTodos } = useFormTodos(handleNewTodo)

    return (
        <form onSubmit={onSubmitTodos}>
            <input type='text' placeholder='Que hay que hacer?' value={input} onChange={onInputChange} />
            <button type='submit'>Agregar</button>
        </form>
    )
}
