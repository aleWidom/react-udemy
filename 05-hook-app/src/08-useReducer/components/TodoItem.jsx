


export const TodoItem = ({ description, id, done, handleDeleteTodo, handleUpdatedDone }) => {
    return (
        <li >
            <span onClick={()=> handleUpdatedDone(id)}>{description}</span>
            <button onClick={()=> handleDeleteTodo(id)}>Borrar</button>
        </li>
    )
}
