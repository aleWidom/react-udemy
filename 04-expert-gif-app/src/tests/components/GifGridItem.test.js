
import { render, screen } from "@testing-library/react"
import { GifGridItem } from '../../components';


describe('GifGridItem', () => {

  const title = 'title';
  const url = 'http://'

  test('Snapshot', () => {
    const { container } = render(<GifGridItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  });

  test('DebeMostrarLaImagenConElUrlYAltIndicado', () => {
    render(<GifGridItem title={title} url={url} />)
  /* expect(screen.getByRole('img').alt).toBe(title)
     expect(screen.getByRole('img').src).toBe(url), otra forma más elegante a continuación */
    const {src, alt} = screen.getByRole('img')
    expect(alt).toBe(title)
    expect(src).toBe(url)
  });
})