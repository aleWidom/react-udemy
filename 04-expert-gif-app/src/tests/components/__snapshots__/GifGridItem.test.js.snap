// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__bounce"
  >
    <img
      alt="title"
      src="http://"
    />
  </div>
</div>
`;
